lib/purchases_flutter.dart

```diff
@@ -894,6 +894,88 @@ class Purchases {
     return PromotionalOffer.fromJson(Map<String, dynamic>.from(result));
   }
 
+  /// iOS only, requires iOS 18.0 or greater with StoreKit 2.
+  ///
+  /// Use this function to retrieve the eligible [WinBackOffer]s that a subscriber
+  /// is eligible for for a given [StoreProduct].
+  ///
+  /// [product] The [StoreProduct] the user intends to purchase.
+  static Future<List<WinBackOffer>> getEligibleWinBackOffersForProduct(
+    StoreProduct product,
+  ) async {
+    final result =
+        await _channel.invokeMethod('eligibleWinBackOffersForProduct', {
+      'productIdentifier': product.identifier,
+    });
+
+    return (result as List)
+        .map((e) => WinBackOffer.fromJson(Map<String, dynamic>.from(e)))
+        .toList();
+  }
+
+  /// iOS only, requires iOS 18.0 or greater with StoreKit 2.
+  ///
+  /// Use this function to retrieve the eligible [WinBackOffer]s that a subscriber
+  /// is eligible for for a given [Package].
+  ///
+  /// [package] The [Package] the user intends to purchase.
+  static Future<List<WinBackOffer>> getEligibleWinBackOffersForPackage(
+    Package package,
+  ) async =>
+      getEligibleWinBackOffersForProduct(package.storeProduct);
```

Comment:
We intentionally use the `getEligibleWinBackOffersForProduct` function here so that the offer caching only needs to be implemented once in the PHC layer